README.md

```diff
@@ -19,3 +25,98 @@ Documentation can be generated with [ExDoc](https://github.com/elixir-lang/ex_do
 and published on [HexDocs](https://hexdocs.pm). Once published, the docs can
 be found at <https://hexdocs.pm/wanda>.
 
+## Developing Checks
+
+Wanda architecture aims to simplify [testing Checks Executions](#testing-executions) and [adding new ones](#adding-new-checks). 
+
+### Infrastructure
+
+A docker-compose setup is provided to enable seamless experience with the system.
+
+#### Starting a local environment
+Start the required infrastructure (see [docker-compose.checks.yaml](./docker-compose.checks.yaml))
+
+```bash
+$ docker-compose -f docker-compose.checks.yaml up -d
+```
+
+This starts:
+- wanda, the engine
+- a database where execution informations are stored
+- a message broker for infromation exchange between the targets and wanda
+
+Wanda is exposed on port `4000` and the API documentation is available at http://localhost:4000/swaggerui
+
+![OpenAPI Spec](./guides/img/api_spec.png)
+
+**Note** that the [message broker](https://www.rabbitmq.com/) **must** be reachable by wanda and all the targets.
+
+### Testing Executions
+
+With a runnig setup it is possible to easily test Checks and their Execution by:
+- consulting the catalog
+- triggering a Checks Execution
+- checking up the state of the triggered execution
+
+#### **Consulting the catalog**
+
+Available Checks are part of the **Catalog**, and they can be retrieved by accessing the dedicated API
+```bash
+curl -X 'GET' \
+  'http://localhost:4000/api/checks/catalog' \
+  -H 'accept: application/json'
+```
+
+#### **Triggering a Checks Execution**
```

Comment:
I wouldn't use the `Trigger` word that much. If the controller, endpoint and all around the code says `start execution`, we could simply use that word.
`Starting a Checks Execution` if you will.
But well, this is personal opinion hehe

Reply:
makes sense, thanks!